ir/src/main/scala/org/scalajs/core/ir/Types.scala

```diff
@@ -84,13 +84,14 @@ object Types {
   case object NullType extends Type
 
   /** Reference types (allowed for classOf[], is/asInstanceOf[]). */
-  sealed abstract class ReferenceType extends Type
+  sealed trait ReferenceType
```

Comment:
I would argue that it would be *more* fragile, because it would lead to people assuming that it is always safe and meaningful to convert a `ReferenceType` into a `Type`, which it is not.

The *proper* `toType` conversion would require a predicate `isJSType: /* className: */ String => Boolean` to know whether `ClassType(className)` has to erase to `AnyType` or not.